crates/bevy_core/src/time/time.rs

```diff
@@ -1,26 +1,41 @@
 use bevy_ecs::system::ResMut;
 use bevy_utils::{Duration, Instant};
 
-/// Tracks elapsed time since the last update and since the App has started
+use crate::FixedTimestepState;
+
+/// Tracks time elapsed since the previous update and since the app was started.
 #[derive(Debug, Clone)]
 pub struct Time {
     delta: Duration,
-    last_update: Option<Instant>,
-    delta_seconds_f64: f64,
     delta_seconds: f32,
-    seconds_since_startup: f64,
+    delta_seconds_f64: f64,
+    fixed_delta: Duration,
+    fixed_delta_seconds: f32,
+    fixed_delta_seconds_f64: f64,
+    relative_speed: f32,
+    time_since_startup: Duration,
+    first_update: Option<Instant>,
+    last_update: Option<Instant>,
     startup: Instant,
 }
 
 impl Default for Time {
-    fn default() -> Time {
-        Time {
+    fn default() -> Self {
+        // 60Hz is a popular number, but it can't be expressed as an exact float.
+        // Power-of-two frequencies (i.e. 32Hz, 64Hz, 128Hz) have better numeric stability.
+        let fixed_delta = Duration::from_secs_f32(1.0 / 64.0);
```

Comment:
If we go for numerical stability (which is the right thing to do), should we really store inside a `Duration`, which is "composed of a whole number of seconds and a fractional part represented in nanoseconds"? It seems `f64` would be a better candidate to avoid unnecessary composition/split operations that may introduce errors.

Reply:
Adding `Duration` values is [exact][1] with nanosecond precision (so is multiplying them by integers). That's kinda the point. Using an `f64` for the accumulator and timestep would introduce rounding errors (albeit very small ones).

I tested some random float values in this [playground][2], but I didn't see any errors converting the `Duration` back into an `f32` (which is why the `Time::set_steps_per_second` takes an `f32`).

I wrote the comment about numerical stability and power-of-two frequencies because I expect the most users to do stuff like this, where adding or multiplying by a power-of-two would have no rounding errors.

```rust
fn integrate_velocity(
    time: Res<Time>,
    query: Query<(&mut Transform, &Velocity)>,
) {
    for (mut transform, velocity) in query.iter_mut() {
        *transform.translation += *velocity * time.fixed_delta_seconds();
    }
}
```
[1]: https://doc.rust-lang.org/stable/src/core/time.rs.html#469-485
[2]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=873196ed2d887a0cff0523eb6cd64e2e